mapreader/spot_text/runner_base.py

```diff
@@ -406,35 +458,107 @@ def convert_to_coords(
                         - yy * self.parent_df.loc[parent_id, "dlat"]
                     )
 
+                    parent_polygon_geo = Polygon(zip(xx, yy)).buffer(0)
                     crs = self.parent_df.loc[parent_id, "crs"]
 
-                    parent_polygon_geo = Polygon(zip(xx, yy)).buffer(0)
                     self.geo_predictions[parent_id].append(
-                        [parent_polygon_geo, crs, *instance[1:]]
+                        GeoPrediction(
+                            pixel_geometry=instance.pixel_geometry,
+                            score=instance.score,
+                            text=instance.text,
+                            patch_id=instance.patch_id,
+                            geometry=parent_polygon_geo,
+                            crs=crs,
+                        )
                     )
 
         if return_dataframe:
-            return self._dict_to_dataframe(self.geo_predictions, geo=True, parent=True)
+            return self._dict_to_dataframe(self.geo_predictions)
         return self.geo_predictions
 
     def save_to_geojson(
         self,
-        save_path: str | pathlib.Path,
+        path_save: str | pathlib.Path,
```

Comment:
I think this one should just be called `path`? Since save is in the method name, we don't need to repeat it in the keyword arguments?